### PR TITLE
Updated omnibus `postinst` script to symlink to appbundle created binstubs

### LINKED ISSUE #2727: InSpec 2.0.17 for macOS 10.13 doesn't work (Gem::GemNotFoundException)
### Description

After installing InSpec from macOS 10.13/64bits download, I can't launch inspec in command line, I get an error `Gem::GemNotFoundException`

### InSpec and Platform Version

- macOS 10.13.3
- https://packages.chef.io/files/stable/inspec/2.0.17/mac_os_x/10.13/inspec-2.0.17-1.dmg

### Replication Case

- Download https://packages.chef.io/files/stable/inspec/2.0.17/mac_os_x/10.13/inspec-2.0.17-1.dmg
- Install InSpec
- Launch terminal
- type `inspec`

### Stacktrace

```
/opt/inspec/embedded/lib/ruby/2.4.0/rubygems.rb:271:in `find_spec_for_exe': can't find gem inspec (>= 0.a) (Gem::GemNotFoundException)
	from /opt/inspec/embedded/lib/ruby/2.4.0/rubygems.rb:299:in `activate_bin_path'
	from /usr/local/bin/inspec:23:in `<main>'
```

### PR DESCRIPTION
**Pull request by miah**
The binstubs created by rubygems, and linked to by our omnibus postinst script error when they encounter GEM_HOME or GEM_PATH set to anything outside of the `/opt/inspec/embedded` directory.

On my laptop with chruby the environment variables GEM_HOME and GEM_PATH are set when I create a new shell.  When I invoke InSpec via `/usr/local/bin/inspec` or `/opt/inspec/embedded/bin/inspec` I get the following error:

```
$ /usr/local/bin/inspec version
/opt/inspec/embedded/lib/ruby/2.4.0/rubygems.rb:271:in `find_spec_for_exe': can't find gem inspec (>= 0.a) (Gem::GemNotFoundException)
	from /opt/inspec/embedded/lib/ruby/2.4.0/rubygems.rb:299:in `activate_bin_path'
	from /usr/local/bin/inspec:23:in `<main>'
```

This error persists until I `unset GEM_HOME` and `unset GEM_PATH`.

```
$ unset GEM_PATH GEM_HOME
$ /usr/local/bin/inspec version
2.0.16
```

The inspec binstub created by appbundler correctly deals with GEM_HOME and GEM_PATH.

```
$ /opt/inspec/bin/inspec version
2.0.16
```

Because our omnibus package already uses [appbundler](https://github.com/chef/appbundler), the fix is to change our symlink to use it rather than the rubygems binstub.

This problem exists in older 1.x packaged releases as well.

Fixes https://github.com/chef/inspec/issues/2727

### CODE REVIEW

File: omnibus/package-scripts/inspec/postinst [hunk 1]
```diff
@@ -5,7 +5,7 @@
 #
 
 PROGNAME=`basename $0`
-INSTALLER_DIR=/opt/inspec/embedded
```
Comment (adamleff) [inline omnibus/package-scripts/inspec/postinst:8]: To keep this consistent with other omnibus projects, I would leave this as `INSTALLER_DIR` and just change the value to `/opt/inspec`

**Review by adamleff — DISMISSED** [review]
This looks like a fine change, and matches what we do in other projects (see https://github.com/chef/chef/blob/master/omnibus/package-scripts/chef/postinst#L14 for example).

For consistency with other projects, I'd like to see the var name stay the same. Also, before we approve and proceed, have we done an ad-hoc Jenkins build from this branch and tested the resulting package on macOS and one other OS (such as Ubuntu) just to make sure this succeeds?

Another thing to think about (outside of this PR) is a change to the Jenkins "test" job to do a `GEM_HOME=/nonexistent/path GEM_PATH=/nonexistent/path /usr/local/bin/inspec version` or whatever the symlink path is to ensure the package works properly.

File: omnibus/package-scripts/inspec/postinst (hunk 1 above)
Comment (miah) [reply omnibus/package-scripts/inspec/postinst:8]: Thanks for the feedback @adamleff =)

**Comment by miah** [comment]
Excellent comment about Jenkins test as well, I'll add one!

**Comment by miah** [comment]
I added two scripts to this PR which will be called by Jenkins during the verify stage.

**Review by jerryaldrichiii — APPROVED** [review]
Looks good to me. Great work @miah!

**Comment by miah** [comment]
@adamleff What do you think friend? =)

**Review by adamleff — APPROVED** [review]
I dig it!

**Review by jquick — APPROVED** [review]
Thanks @miah !

**Pull request merged**